sdk/servicebus/azure_messaging_servicebus/src/client.rs

```diff
@@ -0,0 +1,552 @@
+    pub api_version: String,
+
+    /// Core client configuration options.
+    pub client_options: ClientOptions,
+
+    /// Application ID that will be passed to the namespace.
+    ///
+    /// This optional identifier is passed to the Service Bus namespace during connection establishment
+    /// and can be used for diagnostic purposes. It follows the same pattern as the Go SDK's ApplicationID.
+    pub application_id: Option<String>,
+}
+
+impl Default for ServiceBusClientOptions {
+    fn default() -> Self {
+        Self {
+            api_version: "2017-04".to_string(), // Default Service Bus API version
+            client_options: ClientOptions::default(),
+            application_id: None,
+        }
+    }
+}
+
+/// Options for creating a sender.
+#[derive(Clone, Default)]
+pub struct CreateSenderOptions {
+    // Place holder for future options
+}
+
+/// Options for creating a receiver.
+#[derive(Clone)]
+pub struct CreateReceiverOptions {
+    /// The receive mode for the receiver.
+    pub receive_mode: ReceiveMode,
+    /// The sub-queue to target (e.g., dead letter queue).
+    pub sub_queue: Option<SubQueue>,
+}
+
+impl Default for CreateReceiverOptions {
+    fn default() -> Self {
+        Self {
+            receive_mode: ReceiveMode::PeekLock,
+            sub_queue: None,
+        }
+    }
+}
+
+/// A client for interacting with Azure Service Bus.
+pub struct ServiceBusClient {
+    connection: Arc<AmqpConnection>,
+    namespace: String,
+    options: ServiceBusClientOptions,
+    authorizer: Option<Arc<Authorizer>>,
+}
+
+impl ServiceBusClient {
+    /// Creates a helper function to build AmqpConnectionOptions from ServiceBusClientOptions.
+    fn build_connection_options(
+        options: &ServiceBusClientOptions,
+    ) -> Option<AmqpConnectionOptions> {
+        if let Some(application_id) = &options.application_id {
+            let mut properties = AmqpOrderedMap::new();
+            properties.insert(
+                AmqpSymbol::from("user-agent"),
+                AmqpValue::from(application_id.clone()),
+            );
+            Some(AmqpConnectionOptions {
+                properties: Some(properties),
+                ..Default::default()
+            })
+        } else {
+            None
+        }
+    }
+    /// Returns a builder which can be used to create a new instance of [`ServiceBusClient`].
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use azure_messaging_servicebus::ServiceBusClient;
+    /// use azure_identity::DeveloperToolsCredential;
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    ///     let credential = DeveloperToolsCredential::new(None)?;
+    ///     let client = ServiceBusClient::builder()
+    ///         .open("my-servicebus.servicebus.windows.net", credential.clone()).await?;
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn builder() -> ServiceBusClientBuilder {
```

Comment:
*Client* builders should go in `clients`. Seems we don't really have a guideline that spells it out, but I could add one. But I would hope it would be obvious enough: client builds in `clients`, model builders (unlikely, but possible) in `models`.